<!--
Choose a launcher of gripper for a robot from the robot code.
Since this uses eval in if, it works with Kinetic+.
-->
<launch>

  <arg name="robot_code" default="" />
  <arg name="dxldev" default="/dev/ttyUSB0" />
  <arg name="is_sim" default="false" />
  <arg name="gripper_modbus_uri" default="10.10.6.207" />
  <arg name="geh_auto_mode_sw" default="true" />

  <arg name="robot_code_nosim" value="$(eval robot_code.replace('_SIM','') if robot_code.endswith('_SIM') else robot_code)" />

  <!--NOTE:Changed to use the gripper ThGA with the newer protocol (A) even for
      an older robot_code ending with ThG.
      In case we want to use both versions, replace this by RHP12RNGripper. -->
  <arg name="gripper_type" value="$(eval
        'DxlG' if robot_code_nosim.endswith('DxlG') else
        'ThGA' if robot_code_nosim.endswith('ThG') else
        'ThGA' if robot_code_nosim.endswith('ThGA') else
        'DxlpO2_Straight1' if robot_code_nosim.endswith('DxlpO2_Straight1') else
        'DxlpO2_SRound1'   if robot_code_nosim.endswith('DxlpO2_SRound1')   else
        'DxlpO2_Fork1'     if robot_code_nosim.endswith('DxlpO2_Fork1')     else
        'DxlpY1'           if robot_code_nosim.endswith('DxlpY1')           else
        'DxlO3'            if robot_code_nosim.endswith('DxlO3')            else
        ''.join(['GEH60']+robot_code_nosim.split('GEH60')[1:]) if 'GEH60' in robot_code_nosim else
        ''  )" />

  <group if="$(eval gripper_type!='')">
    <include file="$(find ay_util)/launch/gripper_selector.launch">
      <arg name="gripper_type" value="$(arg gripper_type)" />
      <arg name="dxldev" value="$(arg dxldev)" />
      <arg name="is_sim" value="$(arg is_sim)" />
      <arg name="modbus_uri" value="$(arg gripper_modbus_uri)" />
      <arg name="geh_auto_mode_sw" value="$(arg geh_auto_mode_sw)" />
    </include>
  </group>

</launch>
